content/3_Silver/Prefix_Sums.mdx

```diff
@@ -307,3 +307,122 @@ for i in range(Q):
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer correct>
+      15; 17
+      <Quiz.Explanation>
+        Correct! The sum of the values of some subarray $[l, r]$ is determined by the formula $P[r] - P[l - 1]$.
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      15; 21
+      <Quiz.Explanation>
+        Incorrect. Recall that to get the sum of some range of values $[l, r]$, we need to calculate $P[r] - P[l - 1]$ instead of $P[r] + P[l - 1]$. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+  </Quiz.Question>
+  <Quiz.Question>
+    **Challenge Problem**: With the idea of prefix sums in mind, at first glance it may seem logical to apply this method to products. Instead of adding, we multiply, and instead of subtracting, we divide. Assume that we calculate the answer modulo a large prime ($10^9 + 7$, for instance). What additional challenges may we face?
+    <Quiz.Answer>
+      **Integer Overflow**: Even if we use modulo, the numbers will grow too quickly when multiplying, to the point where they may not fit within a 64-bit integer.
+      <Quiz.Explanation>
+        Incorrect. We can calculate intermediate modulos, which will mitigate this problem. This property is a result of the distribute property of modulo which states that $(a \cdot b) \mod m = (a \mod m) \cdot (b \mod m_ \mod m$. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      **Time Limit**: Multiplying, dividing, and modulo are relatively slow operations, and this will make the prefix product just as fast as iterating through the array.
+      <Quiz.Explanation>
+        Incorrect. While it is true that multiplying and dividing are relatively slow, this difference is hardly enough to impact the runtime by a significant amount. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer>
+      **Memory Limit**: When we switch to products, the memory goes from $\mathcal{O}(n)$ to $\mathcal{O}(n^2)$, which will result in MLE for $n \geq 10^5$. 
+      <Quiz.Explanation>
+        Incorrect. The memory remains the same regardless of whether we use sum or product. 
+      </Quiz.Explanation>
+    </Quiz.Answer>
+    <Quiz.Answer correct>
+      **Division under Modulo**: We cannot directly use division as a substitute for subtraction, because division is not distributive under modulo. 
+      <Quiz.Explanation>
+        Correct! This is to say, for some $a, b$ such that $a$ is divisible by $b$, 
+$$
+\frac a b \mod m \neq \frac {a \mod m} b \mod m
+$$
+Because the problem statement specifies that $m$ is prime, we can circumvent this problem by multiplying by $b^{-1} \mod m$ instead of dividing by $b$. For more information about this, see the [Modular Arithmetic Module](https://usaco.guide/gold/modular?lang=cpp). Note that knowledge of number theory  is generally not needed to score well on silver. 
```

Comment:
Doesn't multiplying by an inverse count as modular division? That's the way I have it set up in my [template](https://github.com/bqi343/USACO/blob/52890a77f66f8b779ba2b1da0460b85dd9f38013/Implementations/content/number-theory%20(11.1)/Modular%20Arithmetic/ModInt.h#L39) ...

Reply:
bad phrasing, is the fixed version better?

Reply:
I don't see how you addressed the comment. Since modular division exists, why is the last answer the correct one?

Reply:
I changed the problem statement such that the modulo doesn't always have to prime. Does this work?